From 3ab6a28796f1c58f982f5380697f7de343ca2ba2 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Sun, 20 Nov 2022 17:45:11 +0100
Subject: [PATCH 03/14] nl80211: remove ancient wpa_supplicant ctrl socket path

It's been +8 years.

See
"735a03be: iwinfo: adjust for changed wpa_supplicant control socket path"
on the main repository.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 iwinfo_nl80211.c | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -935,8 +935,7 @@ static int nl80211_wpactl_connect(const
 
 	remote.sun_family = AF_UNIX;
 	remote_length = sizeof(remote.sun_family) +
-		sprintf(remote.sun_path, "/var/run/wpa_supplicant-%s/%s",
-		        ifname, ifname);
+		sprintf(remote.sun_path, "/var/run/wpa_supplicant/%s", ifname);
 
 	/* Set client socket file permissions so that bind() creates the client
 	* socket with these permissions and there is no need to try to change
@@ -958,14 +957,8 @@ static int nl80211_wpactl_connect(const
 
 	if (connect(sock, (struct sockaddr *)&remote, remote_length))
 	{
-		remote_length = sizeof(remote.sun_family) +
-			sprintf(remote.sun_path, "/var/run/wpa_supplicant/%s", ifname);
-
-		if (connect(sock, (struct sockaddr *)&remote, remote_length))
-		{
-			close(sock);
-			return -1;
-		}
+		close(sock);
+		return -1;
 	}
 
 	local->sun_family = AF_UNIX;
